packages/elastic-apm-synthtrace/src/lib/apm/client/apm_synthtrace_kibana_client.ts

```diff
@@ -40,4 +40,36 @@ export class ApmSynthtraceKibanaClient {
       }
     });
   }
+
+  async discoverLocalKibana() {
+    return await fetch('http://localhost:5601', {
+      method: 'HEAD',
+      follow: 1,
+      redirect: 'manual',
+    }).then((res) => {
+      const kibanaUrl = res.headers.get('location');
+      this.logger.info(`Discovered local kibana running at: ${kibanaUrl}`);
+      return kibanaUrl;
+    });
+  }
+
+  async installApmPackage(kibanaUrl: string, version: string, username: string, password: string) {
+    const response = await fetch(kibanaUrl + '/api/fleet/epm/packages/apm/' + version, {
```

Comment:
Does fleet _usually_ release the apm package  in correspondence to elasticsearch version? 
For example, `/api/fleet/epm/packages/apm/v8.2.0` doesn't exists instead is named `v8.2.0-dev3`

Reply:
It does but this needs to be HEAD aware where we are working ahead of the next release. 

Amended the PR with [`e436669` (#127257)](https://github.com/elastic/kibana/pull/127257/commits/e43666991bbf4ef5da7e8aa60e36c5104315bde2) to make the version selection of the package go through an actual resolving process.

Reply:
nice